apps/dashboard/src/contract-ui/tabs/tokens/components/transfer-button.tsx

```diff
@@ -1,49 +1,160 @@
-import { Icon, useDisclosure } from "@chakra-ui/react";
+import {
+  Sheet,
+  SheetContent,
+  SheetFooter,
+  SheetHeader,
+  SheetTitle,
+} from "@/components/ui/sheet";
+import { FormControl, Icon, Input } from "@chakra-ui/react";
+import { TransactionButton } from "components/buttons/TransactionButton";
+import { SolidityInput } from "contract-ui/components/solidity-inputs";
+import { useTrack } from "hooks/analytics/useTrack";
+import { useTxNotifications } from "hooks/useTxNotifications";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
 import { FiSend } from "react-icons/fi";
-import type { ThirdwebContract } from "thirdweb";
-import { balanceOf } from "thirdweb/extensions/erc20";
-import { useActiveAccount, useReadContract } from "thirdweb/react";
-import { Button, Drawer } from "tw-components";
-import { TokenTransferForm } from "./transfer-form";
+import { type ThirdwebContract, ZERO_ADDRESS } from "thirdweb";
+import { balanceOf, decimals, transfer } from "thirdweb/extensions/erc20";
+import {
+  useActiveAccount,
+  useReadContract,
+  useSendAndConfirmTransaction,
+} from "thirdweb/react";
+import {
+  Button,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+} from "tw-components";
 
 interface TokenTransferButtonProps {
   contract: ThirdwebContract;
 }
 
+const TRANSFER_FORM_ID = "token-transfer-form";
+
 export const TokenTransferButton: React.FC<TokenTransferButtonProps> = ({
   contract,
   ...restButtonProps
 }) => {
-  const { isOpen, onOpen, onClose } = useDisclosure();
   const address = useActiveAccount()?.address;
   const tokenBalanceQuery = useReadContract(balanceOf, {
     contract,
     address: address || "",
     queryOptions: { enabled: !!address },
   });
-
+  const trackEvent = useTrack();
+  const form = useForm({ defaultValues: { amount: "0", to: "" } });
   const hasBalance = tokenBalanceQuery.data && tokenBalanceQuery.data > 0n;
+  const { onSuccess, onError } = useTxNotifications(
+    "Successfully transferred tokens",
+    "Failed to transfer tokens",
+    contract,
+  );
+
+  const decimalsQuery = useReadContract(decimals, { contract });
+  const sendConfirmation = useSendAndConfirmTransaction();
+  const [open, setOpen] = useState(false);
 
   return (
     <>
-      <Drawer
-        allowPinchZoom
-        preserveScrollBarGap
-        size="lg"
-        onClose={onClose}
-        isOpen={isOpen}
-      >
-        <TokenTransferForm contract={contract} />
-      </Drawer>
       <Button
         colorScheme="primary"
         leftIcon={<Icon as={FiSend} />}
         {...restButtonProps}
-        onClick={onOpen}
+        onClick={() => setOpen(true)}
         isDisabled={!hasBalance}
       >
         Transfer
       </Button>
+      <Sheet open={open} onOpenChange={setOpen}>
```

Comment:
instead of manually mutating the sheet state should be able to use `<SheetTrigger asChild><Button>...` here?

Reply:
If we do that we cannot close the Sheet programmatically when the transaction is executed successfully (which is what the "old" Drawer does).

So as I understand it, there are 2 ways:
- SheetTrigger -> Can't close it using code
- <Sheet open={open} onOpenChange={setOpen}> | Can close it using code